packages/design-system/src/components/NumericInputField/NumericInput.js

```diff
@@ -122,6 +122,14 @@ class NumericInput extends React.PureComponent {
     this.removeKeyboardEventListeners();
   };
 
+  parseValue = () => {
+    const { value } = this.props;
+    if(isNaN(value)) {
```

Comment:
strictly this isn't correct, a string is never a NaN (a really bad name for the concept of "not a Real number"), though this will look like it works since value will be coerced into a number before being tested.
@walaszczykm 's suggestion didn't have this issue, and you should probably use Number.isNaN instead of the global isNaN to avoid the coercion issue I mentioned above.